sig-architecture/generics.md

```diff
@@ -0,0 +1,38 @@
+# Background
+
+In general, Kubernetes has not restricted using new Go features, we've quickly
+adopted new standard library types and methods and will continue to do so.
+
+Generally the latest stable go release is in use on the main development branch.
+This includes all of the staging libraries (client-go etc.) that originate in the
+main [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) repository.
+
+In Kubernetes v1.24 we shipped Go 1.18 (which adds support for generics)
+rather late in the release cycle, so we temporarily prohibited using generics
+in case we ran into other issues and needed to roll back to unblock the release.
+Now that v1.24.0 is out, use of generics should be unrestricted.
```

Comment:
nit: this says "[...] use of generics should be unrestricted" but then below we now say "Generics may be used in Kubernetes starting in v1.25, with the following restrictions applying only until v1.24 is [out of support]"

may simply change this to "[...] use of generics should be allowed."

Reply:
done.